modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/MuleExtensionAnnotationParser.java

```diff
@@ -154,23 +158,15 @@ public static Extension getExtension(Class<?> extensionType)
 
     private static void parseGroupParameters(MetadataType parameterType, List<ParsedParameter> parsedParameters, ClassTypeLoader typeLoader)
     {
-        for (Field field : IntrospectionUtils.getParameterFields(JavaTypeUtils.getType(parameterType)))
-        {
-            if (field.getAnnotation(org.mule.runtime.extension.api.annotation.ParameterGroup.class) != null)
-            {
-                parseGroupParameters(getFieldMetadataType(field, typeLoader), parsedParameters, typeLoader);
-            }
-            else
-            {
-                ParsedParameter parsedParameter = doParseParameter(field.getName(),
-                                                                   getFieldMetadataType(field, typeLoader),
-                                                                   toMap(field.getAnnotations()), typeLoader.getClassLoader());
-                if (parsedParameter != null)
-                {
-                    parsedParameters.add(parsedParameter);
-                }
-            }
-        }
+        Stream<Field> declaredFieldsStream = Arrays.stream(getType(parameterType).getDeclaredFields());
```

Comment:
this code is considerably less efficient than the old one since you're iterating the collection many times. Streams are not a silver bullet and in this case I think that is best to add the new logic into the old for